src/Symfony/Component/ObjectMapper/Metadata/ObjectMapperMetadataFactoryInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Metadata;
+
+/**
+ * Factory to create Mapper metadata.
+ *
+ * @experimental
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+interface ObjectMapperMetadataFactoryInterface
+{
+    /**
+     * @param array<string, mixed> $context
+     *
+     * @return list<Mapping>
+     */
+    public function create(object $object, ?string $property = null, array $context = []): array;
```

Comment:
@soyuka Shouldn't a `@throws MappingException` be added to this interface too since ReflectionObjectMapperMetadataFactory are throwing some ? I can do the PR if you agree with